cnd/src/config/sanity.rs

```diff
@@ -0,0 +1,31 @@
+use crate::config::{Bitcoind, Parity};
+
+pub trait Sanity {
```

Comment:
The verification is going to require network communication, which is async code. This trait doesn't account for that at the moment.

Also, it uses types from the `config` folder, which are meant to only act as the serialization/deserialization layer of the configuration system.

Currently, we sometimes re-use types from the `file` module in the `settings` module. This only happens for types that are literally the same, so we don't have to duplicate them unncessarily.

With this ticket, I think you will have to break that re-use and introduce specific types for each layer (with smarter defaults, that re-use gets impossible):

- `file` should model the config file as closely as possible. Use `Option` types for optional stuff etc, to have an easy time deriving `Deserialize`.
- `settings` should model the settings we want in our application, i.e. no unnecessary `Option`s etc

Reply:
I'm afraid that the this verification would broaden the scope of this ticket unnecessarily. 
I'll create a follow-up ticket for this and focus on the smart-default generation.